### PR TITLE
LPD-27588 Persist new property with boolean value in NotificationRecipientSetting table

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPD-27587](https://liferay.atlassian.net/browse/LPD-27587)
[LPD-27588](https://liferay.atlassian.net/browse/LPD-27588)

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:sf

File: modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/EditNotificationTemplate.tsx [hunk 1]
```diff
@@ -136,21 +136,27 @@ export default function EditNotificationTemplate({
...
+				singleRecipient: recipient.singleRecipient,
+				to: recipient.to,
+			} as EmailRecipients;
+
+			if (Liferay.FeatureFlags['LPD-21580']) {
+				newRecipient = {
+					...recipient,
+					usersLanguage: recipient.usersLanguage,
+				};
+			}
```
Comment (rebsilva) [inline modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/EditNotificationTemplate.tsx:155]: here you are going to lose everything you put inside `newRecipient` when you do the `...recipient`, is this really expected?

**Review by rebsilva — COMMENTED** [review]
hi @pedro-oliveira446! just made a comment to confirm a behavior, everything else LGTM!

File: modules/apps/notification/notification-api/src/main/java/com/liferay/notification/constants/NotificationRecipientSettingConstants.java [hunk 2]
```diff
@@ -34,6 +34,8 @@ public class NotificationRecipientSettingConstants {
 
 	public static final String NAME_USER_SCREEN_NAME = "userScreenName";
 
+	public static final String NAME_USERS_LANGUAGE = "usersLanguage";
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-api/src/main/java/com/liferay/notification/constants/NotificationRecipientSettingConstants.java:37]: @pedro-oliveira446 I believe we should rename to `useUserLocale`, see this patterns:

https://github.com/liferay/liferay-portal-ee/blob/0978689ba74c32329bc128bf65fcb56d93140d37/modules/apps/segments/segments-service/src/main/java/com/liferay/segments/model/impl/SegmentsEntryModelImpl.java#L658

https://github.com/liferay/liferay-portal-ee/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/document-library-opener/document-library-opener-service/src/main/java/com/liferay/document/library/opener/service/persistence/impl/DLOpenerFileEntryReferencePersistenceImpl.java#L1018

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
hey @pedro-oliveira446 see my comments

thank you!

File: modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/EditNotificationTemplate.tsx (hunk 1 above)
Comment (pedro-oliveira446) [reply modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/EditNotificationTemplate.tsx]: Hey @rebsilva,
good catch, this is not the expected I changed it.

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by MarinhoFeliphe** [comment]
ci:forward

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Comment by MarinhoFeliphe** [comment]
I've triggered forward accidentally, but I'll close the pull request if it is forwarded to Brian.

cc: @rebsilva @pedro-oliveira446

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @pedro-oliveira446 after reading the details of the story, I believe we should make some changes, let's talk about it Tomorrow.

File: modules/apps/notification/notification-api/src/main/java/com/liferay/notification/constants/NotificationRecipientSettingConstants.java [hunk 3]
```diff
@@ -32,6 +32,8 @@ public class NotificationRecipientSettingConstants {
 
 	public static final String NAME_TO_TYPE = "toType";
 
+	public static final String NAME_USE_USER_LOCALE = "useUserLocale";
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-api/src/main/java/com/liferay/notification/constants/NotificationRecipientSettingConstants.java:35]: @pedro-oliveira446 rename to `useMostRelevantLocaleForGuestUsers`, I chose to follow the same pattern used in `maximumFileSizeForGuestUsers`

**Comment by rebsilva** [comment]
hey @pedro-oliveira446, do you want me to review or you will still make changes? please let me know, thanks!

File: modules/apps/notification/notification-api/src/main/java/com/liferay/notification/constants/NotificationRecipientSettingConstants.java (hunk 3 above)
Comment (pedro-oliveira446) [reply modules/apps/notification/notification-api/src/main/java/com/liferay/notification/constants/NotificationRecipientSettingConstants.java]: Done!

**Comment by pedro-oliveira446** [comment]
Hey @rebsilva ,

I need your review again. There has been a change in the business rule, so the default value of the checkbox will now be true. I have updated this accordingly. If you have any further questions, please feel free to let me know.

**Comment by rebsilva** [comment]
nice @pedro-oliveira446, i will be taking a look now, thanks!

**Review by rebsilva — APPROVED** [review]
the FE changes LGTM! :tada:

File: modules/apps/notification/notification-api/src/main/java/com/liferay/notification/constants/NotificationRecipientSettingConstants.java [hunk 4]
```diff
@@ -32,6 +32,9 @@ public class NotificationRecipientSettingConstants {
 
 	public static final String NAME_TO_TYPE = "toType";
 
+	public static final String NAME_USE_MOST_RELEVANT_LOCALE_GUEST_USERS =
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-api/src/main/java/com/liferay/notification/constants/NotificationRecipientSettingConstants.java:35]: Rename to `NAME_USE_MOST_RELEVANT_LOCALE_FOR_GUEST_USERS`

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_9_0/NotificationRecipientSettingUpgradeProcess.java [hunk 5]
```diff
@@ -5,93 +5,23 @@
...
-
-			while (resultSet2.next()) {
-				_insertNotificationRecipientSetting(
-					preparedStatement3, resultSet2);
-			}
-
-			preparedStatement3.executeBatch();
-		}
+	protected String getNotificationRecipientSettingName() {
+		return "singleRecipient";
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_9_0/NotificationRecipientSettingUpgradeProcess.java:19]: Hey @pedro-oliveira446 use the `NotificationRecipientSettingConstants` here

Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_9_0/NotificationRecipientSettingUpgradeProcess.java:19]: Hey @pedro-oliveira446 use the `NotificationRecipientSettingConstants` here

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_2/NotificationRecipientSettingUpgradeProcess.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * @author Pedro Leite
+ */
+public class NotificationRecipientSettingUpgradeProcess
+	extends BaseNotificationRecipientSettingUpgradeProcess {
+
+	@Override
+	protected String getNotificationRecipientSettingName() {
+		return "useMostRelevantLocaleForGuestUsers";
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_2/NotificationRecipientSettingUpgradeProcess.java:19]: Hey @pedro-oliveira446 use the `NotificationRecipientSettingConstants` here

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/upgrade/v3_10_2/test/NotificationRecipientSettingUpgradeProcessTest.java [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+			int rowCount = 0;
+
+			while (resultSet.next()) {
+				rowCount++;
+
+				boolean useMostRelevantLocaleForGuestUsers =
+					resultSet.getBoolean("value");
+
+				Assert.assertFalse(useMostRelevantLocaleForGuestUsers);
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/upgrade/v3_10_2/test/NotificationRecipientSettingUpgradeProcessTest.java:158]: @pedro-oliveira446 put it inline

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/upgrade/v3_10_2/test/NotificationRecipientSettingUpgradeProcessTest.java [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								NotificationRecipientSettingConstants.NAME_TO,
+								RandomTestUtil.randomString() + "@liferay.com"),
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								NotificationRecipientSettingConstants.
+									NAME_SINGLE_RECIPIENT,
+								String.valueOf(
+									RandomTestUtil.randomBoolean()))),
```
Comment (MarinhoFeliphe) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/upgrade/v3_10_2/test/NotificationRecipientSettingUpgradeProcessTest.java:93]: Hey @pedro-oliveira446 is the setting `NAME_SINGLE_RECIPIENT` required?

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:objects

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:forward:force

**Comment by pedro-oliveira446** [comment]
ci:forward:force

**Pull request closed without merging**